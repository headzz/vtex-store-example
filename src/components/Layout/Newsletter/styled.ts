import { colors } from 'configuration'
import styled from 'styled-components'

export const NewsletterStyled = styled.section`
  background-color: ${colors.BLACK};
  padding: 6.4rem 0;
  width: 100%;
`
