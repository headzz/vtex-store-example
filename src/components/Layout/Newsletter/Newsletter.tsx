import { NewsletterStyled } from './styled'

/**
 * @component
 * @name Newsletter
 * @description Defines the container for the layout of newsletter section.
 * @namespace components components/Layout/Newsletter
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */

const Newsletter: React.FC = ({ children }) => {
  return <NewsletterStyled>{children}</NewsletterStyled>
}

export default Newsletter
