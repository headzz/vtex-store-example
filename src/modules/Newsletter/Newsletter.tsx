import Layout from 'components/Layout'

import TextArea from './components/TextArea'
import { ContactContainer } from './styled'

/**
 * @component
 * @name Newsletter
 * @description creates Newsletter component section to use on screen.
 * @namespace components modules/Newsletter
 *
 */
const Newsletter: React.FC = () => (
  <Layout.Newsletter>
    <ContactContainer>
      <TextArea />
    </ContactContainer>
  </Layout.Newsletter>
)

export default Newsletter
