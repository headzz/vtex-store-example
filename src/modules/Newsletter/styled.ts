import styled from 'styled-components'

export const ContactContainer = styled.div`
  display: flex;
  justify-content: center;
  div {
    padding: 0 1rem;
    width: 100%;

    p:first-child {
      margin-bottom: 3.2rem;
      text-align: center;
    }
  }
  @media (min-width: 768px) {
    div {
      padding: 0;
      width: 67.6rem;
    }
  }
`
