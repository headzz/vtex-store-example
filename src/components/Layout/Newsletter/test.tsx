import { render } from '@testing-library/react'

import Newsletter from './Newsletter'

describe('<Newsletter />', () => {
  it('should render component that wraps newsletter on page', () => {
    const { container } = render(<Newsletter />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
