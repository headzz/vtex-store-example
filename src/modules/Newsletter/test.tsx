import { render } from '@testing-library/react'

import Newsletter from './Newsletter'

describe('Newsletter', () => {
  it('match snapshot', () => {
    const { container } = render(<Newsletter />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
