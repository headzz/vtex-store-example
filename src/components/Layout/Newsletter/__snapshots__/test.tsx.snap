// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<Newsletter /> should render component that wraps newsletter on page 1`
] = `
.c0 {
  background-color: #000000;
  padding: 6.4rem 0;
  width: 100%;
}

<section
  class="c0"
/>
`
